strings-csharp/SubstringInString/SubstringInString/Benchmarking.cs

```diff
@@ -0,0 +1,45 @@
+﻿using BenchmarkDotNet.Attributes;
+
+[MemoryDiagnoser]
+[Orderer(BenchmarkDotNet.Order.SummaryOrderPolicy.FastestToSlowest)]
+public class Benchmarking
+{
+    public string str = new("Lorem".ToCharArray()[0], 5);
```

Comment:
@lynda-nneka why did you change the string that will be searched? For the benchmark we would want a "normal" size string and preferably one that contains the sought for text multiple times so multiple indexes are returned. You changed the paragraph of text that was here into a 5 character string containing just the letter 'L'. Here is the result if you print the string that is being created here:
`LLLLL`

Reply:
![Screenshot (42)](https://github.com/CodeMazeBlog/CodeMazeGuides/assets/86114880/8957d8c9-5013-40a2-8213-ae6ac57021b8)
I got a feedback here for it.

Reply:
@lynda-nneka I am sorry that that advice was given. In this context we want a longer string which contains the searched for text in multiple locations. Lorem Ipsum is also a good choice in this context. For the benchmark we want to be returning at least a couple instances of matches, just so that we see the performance of the collection being built as well. Since all of this is part of the benchmark.

One way you could make that text cleaner would be to use a raw string literal, that way you don't need all the plus signs. Since the value is constant, you can also declare it as constant.

Either way you handle it, whether it is to put it in a separate file, add a library like Bogus and generate the string from the Lorem data set, or some other method of string creation, for the benchmark we would want to have at least a paragraph of text. And we definitely want text that contains the searched for text.

Reply:
@jshergal Should I put it back as a paragraph in my code or read it as a text file

Reply:
@lynda-nneka I would just put it as a paragraph in the code, but as a raw string literal, which will make it a bit cleaner.

Reply:
@jshergal Why can't we use something like this in a phase of global setup, or iteration setup:
private string str = "";
private const string ToFind = "Lorem";

[GlobalSetup]
public void Setup()
{
    StringBuilder sb = new StringBuilder();

    sb.Append(ToFind);
    sb.Append(new string('a', 1000));
    sb.Append(ToFind);
    sb.Append(new string('a', 1000));
    sb.Append(ToFind);

    str = sb.ToString();
}

IMO: it's better than putting a string literal that contains 10 lines. In such a way we can construct any string with any length without the necessity of changing the string literal.

Reply:
@OTybolc. 
If the string is nothing but a continuous group of the same character, let's use 'a' as you have listed here, and the string is 1000 in length, all 'a'. What indexes would you expect to be returned if we search for "aaa"? The result would be 2,3,4,...,996. There will be `1000 - searchLength` occurrences.

To exercise, in a benchmark, this code, the better example is a block of "real world" text and a search value that occurs multiple times throughout the text. Thus the _lorem ipsum_ text. It looks "real-worldy" and contains the word "lorem" multiple times. Whether we put it in a file or have it live in the code, doesn't matter so much. I would argue that having the paragraph in the code looks okay and is easily accessible to someone running the code, and easy to change.

So the main issue with the repeated character is now the string is everywhere and there is basically no index where it is not found. Whereas the _lorim ipsum_ text provides a more realistic example case, thus validating the benchmarking conditions.